pkg/securitypolicy/framework.rego

```diff
@@ -90,31 +145,51 @@ mountConstraint_ok(constraint, mount) {
         some constraintOption in constraint.options
         option == constraintOption
     }
+
     every option in constraint.options {
         some mountOption in mount.options
         option == mountOption
     }
 }
 
-mount_ok(container, mount) {
-    some constraint in container.mounts
+mount_ok(mounts, allow_elevated, mount) {
+    some constraint in mounts
     mountConstraint_ok(constraint, mount)
 }
 
-mount_ok(container, mount) {
+mount_ok(mounts, allow_elevated, mount) {
     some constraint in data.defaultMounts
     mountConstraint_ok(constraint, mount)
 }
 
-mount_ok(container, mount) {
-    container.allow_elevated
+mount_ok(mounts, allow_elevated, mount) {
+    allow_elevated
     some constraint in data.privilegedMounts
     mountConstraint_ok(constraint, mount)
 }
 
-mountList_ok(container) {
+mountList_ok(mounts, allow_elevated) {
     every mount in input.mounts {
-        mount_ok(container, mount)
+        mount_ok(mounts, allow_elevated, mount)
+    }
+}
+
+default exec_in_container := {"allowed": false}
+
+exec_in_container := {"matches": matches, "allowed": true} {
+    container_started
+    containers := [container |
+        some container in data.metadata.matches[input.containerID]
+        some process in container.exec_processes
```

Comment:
I don't think we should be locking the `env` and `working_dir` for `exec_in_container` in the policy. If I'm reading CRI code correctly, those will be inherited from the container spec.

Reply:
One thing to consider is to check what CRI exposes for `Exec` and `ExecSync`: https://github.com/kubernetes/cri-api/blob/master/pkg/apis/runtime/v1alpha2/api.proto

Reply:
Should we handle stdio enforcement here as well? e.g. allow/disallow interactive shells?

Reply:
> I don't think we should be locking the env and working_dir for exec_in_container in the policy. If I'm reading CRI code correctly, those will be inherited from the container spec.

That's host side and can't be trusted.

Reply:
> Isn't that host side and can't be trusted?

Right, I'm just talking about what should go into the policy. since the users don't really control the environment for the exec and it's simply inherited from the container spec the exec is running inside, I feel like we don't need to have a policy configuration around it. we should, instead, enforce the environment based on the container config.

Reply:
From my understanding of your statement, that is backwards.  Let's discuss synchronously.

Reply:
Ok, we had the conversation. I understand.

Rather than having policy related to the container include the environment variables and cwd, because those are supposed to be inherited from the container, we will use the ones from the container to do enforcement rather than requiring their duplication in the "container process" definition in policy.